source/Octostache.Tests/ParserFixture.cs

```diff
@@ -39,5 +40,70 @@ public void AListOfVariablesCanBeExtracted()
             result.Should().NotContain("item");
             result.Should().NotContain("item.Address");
         }
+
+        [Fact]
+        public void NestedIndexCanBeExtracted()
+        {
+            var template = "#{Octopus.Action[].Package[containers[1].container].PackageVersion}";
+            var result = TemplateParser.ParseTemplateAndGetArgumentNames(template);
+            result.Should().Contain("[containers[1].container]");
+        }
+        
+        [Fact]
+        public void MultipleNestedIndicesCanBeExtracted()
+        {
+            var template = "#{Octopus.Action[].Package[array[foo].containers[1].container].PackageVersion}";
+            var result = TemplateParser.ParseTemplateAndGetArgumentNames(template);
+            result.Should().Contain("[array[foo].containers[1].container]");
+        }
```

Comment:
Instead of having these tests cases here, I think a more suitable place would be `UsageFixture`.